lib/Reader/abisignature.cpp

```diff
@@ -312,7 +312,7 @@ CallSite ABICallSignature::emitUnmanagedCall(GenIR &Reader, Value *Target,
 
 Value *ABICallSignature::emitCall(GenIR &Reader, Value *Target, bool MayThrow,
                                   ArrayRef<Value *> Args,
-                                  Value *IndirectionCell,
+                                  Value *IndirectionCell, bool IsJmp,
```

Comment:
Given that it looks like we'll need to do most (all?) of the transformations that are conditional on this parameter for tail calls as well, I wonder if it wouldn't be better to split this into `bool IsTailCall, bool MustTailCall` (or `bool IsTailCall, bool IsJmp`). Or maybe even something like:

```
enum CallKind
{
    Normal = 0,
    TailCall = 1,
    MustTailCall =2
}
```